plugins/inputs/gtfs/gtfs.go

```diff
@@ -0,0 +1,305 @@
+	var (
+		work errgroup.Group
+		now  = time.Now()
+	)
+
+	if g.vehiclePositionsReq != nil {
+		work.Go(func() error { return g.gatherVehiclePositions(acc, now) })
+	}
+
+	if g.tripUpdatesReq != nil {
+		work.Go(func() error { return g.gatherTripUpdates(acc, now) })
+	}
+
+	if g.serviceAlertsReq != nil {
+		work.Go(func() error { return g.gatherServiceAlerts(acc, now) })
+	}
+
+	return work.Wait()
+}
+
+func (g *GTFS) gatherVehiclePositions(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.vehiclePositionsReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.Vehicle == nil {
+			continue
+		}
+		v := entity.Vehicle
+
+		b, err := json.Marshal(v)
+		if err != nil {
+			return err
+		}
+
+		fields := map[string]interface{}{
+			"latitude":      v.Position.Latitude,
+			"longitude":     v.Position.Longitude,
+			"bearing":       v.Position.Bearing,
+			"odometer":      v.Position.Odometer,
+			"speed":         v.Position.Speed,
+			"vehicle_id":    v.Vehicle.Id,
+			"vehicle_label": v.Vehicle.Label,
+			"trip_id":       v.Trip.TripId,
+			"direction_id":  v.Trip.DirectionId,
+			"stop_id":       v.StopId,
+			"status":        v.CurrentStatus,
+			"congestion":    v.CongestionLevel,
+			"json":          b,
+		}
+
+		tags := map[string]string{
+			"route_id":     *v.Trip.RouteId,
+			"direction_id": fmt.Sprintf("%d", v.Trip.DirectionId),
+		}
+
+		acc.AddFields("position", fields, tags, time.Unix(int64(*v.Timestamp), 0))
+	}
+
+	return nil
+}
+
+func (g *GTFS) gatherTripUpdates(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.tripUpdatesReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.TripUpdate == nil {
+			continue
+		}
+		u := entity.TripUpdate
+
+		b, err := json.Marshal(u)
+		if err != nil {
+			return err
+		}
+
+		for _, update := range u.StopTimeUpdate {
+			fields := map[string]interface{}{
+				"vehicle_id":    u.Vehicle.Id,
+				"vehicle_label": u.Vehicle.Label,
+				"trip_id":       u.Trip.TripId,
+				"stop_id":       update.StopId,
+				"stop_sequence": update.StopSequence,
+				"arrival":       update.Arrival.Time,
+				"departure":     update.Departure.Time,
+				"delay":         u.Delay,
+				"json":          b,
+			}
+
+			tags := map[string]string{
+				"route_id":     *u.Trip.RouteId,
+				"direction_id": fmt.Sprintf("%d", u.Trip.DirectionId),
+			}
+
+			acc.AddFields("update", fields, tags, time.Unix(int64(*u.Timestamp), 0))
+		}
+	}
+
+	return nil
+}
+
+func (g *GTFS) gatherServiceAlerts(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.serviceAlertsReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.Alert == nil {
+			continue
+		}
+		a := entity.Alert
+
+		b, err := json.Marshal(a)
+		if err != nil {
+			return err
+		}
+
+		fields := map[string]interface{}{
+			"cause": a.Cause.String(),
```

Comment:
Is this the only relevant field?

Reply:
I believe so. I added this measurement for completion. In reality, I expect vehicle positions to be the most useful and commonly used measurement provided by this plugin.

Reply:
Thanks, please mark as resolved.